src/shared/bg_pmove.cpp

```diff
@@ -3446,58 +3399,39 @@ static void PM_Footsteps()
 		{
 			bobmove = 0.3f; // walking bobs slow
 
-			if ( pm->ps->pm_flags & PMF_BACKWARDS_RUN )
+			if ( IsSegmentedModel( pm->ps ) )
 			{
-				if ( IsSegmentedModel( pm->ps ) )
+				PM_ContinueLegsAnim( backrun ? LEGS_BACKWALK : LEGS_WALK );
+			}
+			else
+			{
+				if ( pm->cmd.rightmove > 0 && !pm->cmd.forwardmove )
 				{
-					PM_ContinueLegsAnim( LEGS_BACKWALK );
+					PM_ContinueLegsAnim( NSPA_WALKRIGHT );
 				}
-				else
+				else if ( pm->cmd.rightmove < 0 && !pm->cmd.forwardmove )
 				{
-					if ( pm->cmd.rightmove > 0 && !pm->cmd.forwardmove )
-					{
-						PM_ContinueLegsAnim( NSPA_WALKRIGHT );
-					}
-					else if ( pm->cmd.rightmove < 0 && !pm->cmd.forwardmove )
-					{
-						PM_ContinueLegsAnim( NSPA_WALKLEFT );
-					}
-					else
-					{
-						PM_ContinueLegsAnim( NSPA_WALKBACK );
-					}
+					PM_ContinueLegsAnim( NSPA_WALKLEFT );
 				}
-			}
-			else
-			{
-				if ( IsSegmentedModel( pm->ps ) )
+				else if ( backrun )
 				{
-					PM_ContinueLegsAnim( LEGS_WALK );
+					PM_ContinueLegsAnim( NSPA_WALKBACK );
 				}
 				else
 				{
-					if ( pm->cmd.rightmove > 0 && !pm->cmd.forwardmove )
-					{
-						PM_ContinueLegsAnim( NSPA_WALKRIGHT );
-					}
-					else if ( pm->cmd.rightmove < 0 && !pm->cmd.forwardmove )
-					{
-						PM_ContinueLegsAnim( NSPA_WALKLEFT );
-					}
-					else
-					{
-						PM_ContinueLegsAnim( NSPA_WALK );
-					}
+					PM_ContinueLegsAnim( NSPA_WALK );
 				}
 			}
 		}
 	}
 
-	bobmove *= BG_Class( pm->ps->stats[ STAT_CLASS ] )->bobCycle;
+	classAttributes_t const* pcl = BG_Class( pm->ps->stats[ STAT_CLASS ] );
+	ASSERT( pcl );
```

Comment:
Useless assertion as BG_Class always returns something even if given an out-of-range value.

Reply:
When I get a pointer (and not a reference) from a function, I always consider things can go wrong and thus always put an assert after if I do not have full control on input parameters, just to be sure.
I could remove the assert, but it's likely the best fix here would be to return a reference (which by definition can not be null without being a bug).

Reply:
I suppose doing that (returning a ref instead of a pointer) should be addressed in a different PR (as this one is enormous enough as it is, and this would be kind of out of scope).
Maybe this could be done more generally, though: I think I'll do a cleanup on that matter too, if that's ok.